### ISSUE TITLE
Preloaded .obj mesh fails to load in exported project

### ISSUE THREAD

**Issue by vfig** [godotengine/godot#62067]
### Godot version

v4.0.alpha9.official [fc18891db]

### System information

Windows 10

### Issue description

A script uses `preload("res://icospher.obj")` to preload a mesh, before later putting it on a MeshInstance3D. When running the game from the editor, this works as expected. But after exporting the project (to a .exe in this case), the mesh fails to load, and the console shows the following errors:

```
ERROR: Parser bug (please report): tried to assign unset node without an identifier.
   at: resolve_class_interface (modules/gdscript/gdscript_analyzer.cpp:656)
SCRIPT ERROR: Parse Error: Preload file "res://icospher.obj" does not exist.
          at: GDScript::reload (res://main.gd:3)
ERROR: Method/function failed. Returning: ERR_PARSE_ERROR
   at: reload (modules/gdscript/gdscript.cpp:885)
```

When writing the script, the autocomplete suggested the `"res://icospher.obj"`, so I inferred that was the correct way to reference the resource, even though the actual file loaded is buried in the `.godot/imported` subdirectory.

Using a hex editor, I can see the converted mesh resource is present in the exported .pck, but under the name "res://.godot/imported/icospher.obj-ea174b856465bb90893e7d64ea383677.mesh".

### Steps to reproduce

Run the MRP from the editor. You should see the icosphere in the viewport.

Export the sample project (or use the .pck file in the MRP), and run it. No icosphere.

### Minimal reproduction project

[bug_export_noobj.zip](https://github.com/godotengine/godot/files/8908875/bug_export_noobj.zip)

**Comment by Calinou** [comment]
This is likely because the import remapping for the file isn't included in the exported project for some reason. Does this occur if you instance icospher.obj anywhere in your project *before* exporting it?

**Comment by vfig** [comment]
The export works correctly if I do one of:
- I reference the .obj mesh in the main scene directly.
- I `load()` the .obj mesh in `_ready()` instead of preloading it.
- I save the .obj mesh manually to a .tres file and preload that in the script instead.

But it does not work if I do one of:
- I reference the .obj mesh in a second scene that is otherwise unreferenced
- I reference the .obj mesh in a second scene that is preloaded in the script (but not otherwise referenced)
- I reference the .obj mesh in a second scene that is `load()`-ed in the main scene's `_ready()` (but not otherwise referenced)
- I add `*.obj` to the "Filters to export non-resource files/folders" in the export dialog.

**Comment by Chaosus** [comment]
Can confirm it's a very serious bug - also the textures seems to be affected too (and probably other resource types).

**Comment by akien-mga** [comment]
Duplicate of #56343, it's fixed in 4.0 beta 2.

**Comment by Zireael07** [comment]
What beta 2? O_o
